extensions/azdata/src/test/azdata.test.ts

```diff
@@ -144,23 +145,28 @@ describe('azdata', function () {
 
 async function testLinuxUnsuccessfulUpdate() {
 	const executeSudoCommandStub = sinon.stub(childProcess, 'executeSudoCommand').rejects();
-	const updatePromise = azdata.checkAndUpdateAzdata(oldAzdataMock);
-	await should(updatePromise).be.rejected();
+	const updateDone = await azdata.checkAndUpdateAzdata(oldAzdataMock);
+	Logger.log(`testLinuxUnsuccessfulUpdate: executeSudoCommandStub.callCount: ${executeSudoCommandStub.callCount}`);
```

Comment:
Why are you doing this? Just use console.log directly if you want to log stuff for a test

Reply:
It makes for a central place to turn logging on/off instead of 2 mechanism of product and test logs.

Reply:
As with all the other stuff, I don't really like this pattern though. Why have debug log statements if they aren't going to be something we can easily enable without code changes? Having something like you're saying makes sense but if you're going to be doing stuff like this then we should be doing it in a generic way that makes it easy for others to use too and not just a one-off that only this extension benefits from.

So I'd prefer you remove these from this PR and then make a separate PR to address adding this kind of stuff so we can have a separate discussion about how best to do it.

Reply:
You had said that it is ok if we logged additional output from the test all the time. Is it ok if I added sinon stub for  Logger.log to send the log output to the console and then kept these Logger.log statements? It is a pain to insert these statements into the tests every time I have to look into these test failures. Every refactoring of the code will bring in test failures and I do want to keep those costs manageable. So, I would like you to give me a recommended way for me to keep some test logging that I can check-in now without requiring some additional infrastructure work.

Reply:
Alternatively, I will save off these logging changes in a stash, remove them from this PR to check them in and then bring in some simple proposal to have a well-known published mechanism for our tests to allow tests to turn on/off logging.

Reply:
That sounds perfect. Would be a good topic for a tech sync too once you have a proposal. It's definitely a feature I see value in having since we don't want to always be logging everything. 

(My general idea is probably just having a separate test log that gets saved off somewhere and contains the full verbose logging for the tests)